packages/build/src/plugins/node_version.ts

```diff
@@ -78,6 +98,37 @@ const addPluginNodeVersion = function ({
     'https://answers.netlify.com/t/build-plugin-update-system-node-js-version-upgrade-to-20/108633',
   )}`,
     )
+
+    if (pluginPath) {
+      const pluginDir = dirname(pluginPath)
+      const { packageJson: pluginPackageJson } = await getPackageJson(pluginDir)
+
+      // Ensure Node.js version is compatible with plugin's `engines.node`
+      const pluginNodeVersionRange = pluginPackageJson?.engines?.node
+      if (!pluginNodeVersionRange) {
+        logWarning(
+          logs,
+          "This plugin doesn't have a `package.json` file with an `engines.node` field, so it might not work well with Node.js 20.",
```

Comment:
mhm most plugins probably don't have an engines field 🤔 should we really warn about that?

Reply:
but some do, and we could've caught the whole lighthouse issue had we had this. maybe we should change the wording here?

Reply:
also happy to remove the user-facing warning, and only do the system log

Reply:
Yea I think I would go with system logs (because we can query for it) but no need to warn the customer about it because it's valid to not have an engine field but we don't need to warn them (as they cannot action about it)

Reply:
removed the user-facing log in https://github.com/netlify/build/pull/5497/commits/7b7720580224fa90dd8dcc39fff6d4b4c40a1291.